.github/workflows/ci.yaml

```diff
@@ -139,6 +142,73 @@ jobs:
         if: ${{ always() }}
         run: rm -f cosign.key
 
+  create-and-upload-sbom:
+    name: create-and-upload-sbom
+    needs: [release]
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        repo: ${{ fromJSON(needs.release.outputs.container_repos) }}
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v2.1.5
+        with:
+          go-version: 1.17
+
+      - name: Install cosign
+        uses: sigstore/cosign-installer@v1.4.1
+        with:
+          cosign-release: 'v1.4.1'
+
+      - name: Install Syft
+        uses: anchore/sbom-action/download-syft@v0.6.2-draft
+
+      - name: Checkout
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+
+      - name: Login to Container registries
+        run: |
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u philipssoftware --password-stdin
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login -u ${{ github.actor }} --password-stdin ghcr.io
+
+      - name: Install signing key
+        run: |
+          echo '${{ secrets.COSIGN_PRIVATE_KEY }}' > cosign.key
+
+      - name: Create SBOM from container with Syft
+        run: |
+          syft ghcr.io/philips-labs/slsa-provenance:${{ needs.release.outputs.container_digest }} -o spdx-json > sbom.spdx.json
+
+      - name: Upload Container SBOM to OCI
+        run: |
+          cosign upload blob -f sbom.spdx.json ${{ matrix.repo }}/attestations@${{ needs.release.outputs.container_digest }}
```

Comment:
Why are we uploading this as a `blob`? We can also can use the attestation type `spdx`: 

```
cosign attest --predicate sbom-spdx.json --type spdx ........
```